### ISSUE TITLE
Development Instructions

### ISSUE THREAD

**Issue by synedra-datastax** [HadesArchitect/GrafanaCassandraDatasource#70]
It's not really clear what I'm doing here.  The docker thing doesn't leave me with a docker container, do I do the instructions in the other sections?  We need to have a clear path (do 1 of these).

Docker Way (Recommended)

docker run --rm -v ${PWD}:/opt/gcds -w /opt/gcds node:12 npm install
docker run --rm -v ${PWD}:/opt/gcds -w /opt/gcds node:12 node node_modules/webpack/bin/webpack.js
docker run --rm -v ${PWD}:/go/src/github.com/ha/gcp -w /go/src/github.com/ha/gcp/backend golang go mod vendor
docker run --rm -v ${PWD}:/go/src/github.com/ha/gcp -w /go/src/github.com/ha/gcp golang go build -i -o ./dist/cassandra-plugin_linux_amd64 ./backend
Locally

npm install
webpack
cd backend && go mod vendor
go build -i -o ./dist/cassandra-plugin_linux_amd64 ./backend
Building the backend with TLS support [OPTIONAL]

go get -u github.com/go-bindata/go-bindata/... - downloading the bindata package
Place your tls certificate and tls key into ./backend/creds folder
cd ./backend && go-bindata -o assets.go ./creds && cd .. - move credentials files as a .go files
go build -i -ldflags "-X main.CertPath=/creds/cert_file_name -X main.KeyPath=/creds/key_file_name -X main.InsecureSkipVerify=true" -o ./dist/cassandra-plugin_linux_amd64 ./backend - build binary with required variables filled in. If you'd like to use rootCA, do go build -i -ldflags "-X main.RootCA=/creds/root_ca_file_name" -o ./dist/cassandra-plugin_linux_amd64 ./backend

**Comment by HadesArchitect** [comment]
This part of the instructions needs some clarification indeed. Luckily it isn't for plugin users, only devs. I'll think about how to make it clearer.

**Comment by HadesArchitect** [comment]
> docker thing doesn't leave me with a docker container

Yep, it doesn't. You don't need the container afterward, those are the build steps to prepare the frontend and backend, compile things etc.
